### ISSUE TITLE
cloak of shadows not removing pet magic pin

### ISSUE THREAD

**Issue by oldspices** [gamefreedomgit/Maelstrom#5053]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
cloak of shadows removes all magic-

**How to reproduce:**
hunter pet crab pin on rogue

**How it should work:**
cloak should remove all magic
pin is magic, you can dispel it. 

**Database links:**

**Comment by UltraNix** [comment]
Pin is not magic.
https://www.wowhead.com/spell=50245/pin
`SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 2 (SPELL_DAMAGE_CLASS_MELEE)
PreventionType = 2 (SPELL_PREVENTION_TYPE_PACIFY)
`
Cloak of Shadow is not a dispel effect.

**Comment by oldspices** [comment]
> Pin is not magic. https://www.wowhead.com/spell=50245/pin
> 
> ```
> SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
> DamageClass = 2 (SPELL_DAMAGE_CLASS_MELEE)
> PreventionType = 2 (SPELL_PREVENTION_TYPE_PACIFY)
> ```
> 
> Cloak of Shadow is not a dispel effect.

What???? Cloak of shadow most definitely removes magic and the link you provided clearly states pin as a magical buff- i am flabbergasted.

**Comment by UltraNix** [comment]
Okay - let me explain to you.
As you can see, Pin is considered as physical ability because of its spell school.
![image](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/27b70d7a-0ffd-413d-884c-555d128c279c)
So only other physical buffs/debuffs will affect it.

Now - Cloak of Shadows is not a dispel effect spell per se. You have special effects with dispel, like Purge or Cleanse. These can dispel Pin, because Pin has specific DispelType:
![image](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/935c5fff-60ce-44f0-8212-0fdaea64a0e2)
This is also what you can see in debuff icon - it's an info for dispeleres if that can be dispelled via these effects. Eg. Remove Curse won't work because of wrong dispelType (Remove Curse dispels only curses).

Cloak of Shadow gives you an immunity to all magical spells and effects. Magical mean - to all spells with spell school not physical, eg. Arcane or Nature.
```
ID - 81549 Cloak of Shadows
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 0
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
Aura Id 39 (SPELL_AURA_SCHOOL_IMMUNITY), value = 0, misc = 126 (126), miscB = 0, periodic = 0
```
So it gives you immunity to specific spell schools. Due to its special attribute, CoS gives you also removal of such effects. So CoS removes from you all debuffs with such spell school. 126 is a bitmask of all magical spell schools. Here's the enum:
```

    // 124, not include normal and holy damage
    SPELL_SCHOOL_MASK_SPELL   = (SPELL_SCHOOL_MASK_FIRE   |
                                  SPELL_SCHOOL_MASK_NATURE | SPELL_SCHOOL_MASK_FROST  |
                                  SPELL_SCHOOL_MASK_SHADOW | SPELL_SCHOOL_MASK_ARCANE),
    // 126
    SPELL_SCHOOL_MASK_MAGIC   = (SPELL_SCHOOL_MASK_HOLY | SPELL_SCHOOL_MASK_SPELL),

    // 127
    SPELL_SCHOOL_MASK_ALL     = (SPELL_SCHOOL_MASK_NORMAL | SPELL_SCHOOL_MASK_MAGIC)
```

So TLDR: Purge or Cleanse dispels Pin (due to Dispel type)
CoS will not remove Pin (because of mismatching spell schools).

**Comment by oldspices** [comment]
ngl that makes 0 sense to me

**Comment by UltraNix** [comment]
That's how it works. Blame Blizzard, not me.
CoS does not dispel debufs with Magic dispel type.
CoS does remove auras with magic spell schools.
Pin is a physical school spell.
